src/react/utils.js

```diff
@@ -944,18 +944,23 @@ function applyClonedTemporalAlias(realm: Realm, props: ObjectValue, clonedProps:
     // be a better option.
     invariant(false, "TODO applyClonedTemporalAlias conditional");
   }
-  let temporalBuildNodeEntryArgs = realm.getTemporalBuildNodeEntryArgsFromDerivedValue(temporalAlias);
-  invariant(temporalBuildNodeEntryArgs !== undefined);
-  let temporalArgs = temporalBuildNodeEntryArgs.args;
+  let temporalBuildNodeEntry = realm.getTemporalBuildNodeEntryFromDerivedValue(temporalAlias);
+  if (!(temporalBuildNodeEntry instanceof TemporalObjectAssignEntry)) {
+    invariant(false, "TODO nont TemporalObjectAssignEntry");
+  }
+  invariant(temporalBuildNodeEntry !== undefined);
+  let temporalArgs = temporalBuildNodeEntry.args;
   // replace the original props with the cloned one
   let newTemporalArgs = temporalArgs.map(arg => (arg === props ? clonedProps : arg));
 
+  let preludeGenerator = realm.preludeGenerator;
+  invariant(preludeGenerator !== undefined);
   let temporalTo = AbstractValue.createTemporalFromBuildFunction(
     realm,
     ObjectValue,
     newTemporalArgs,
-    ([methodNode, targetNode, ...sourceNodes]: Array<BabelNodeExpression>) => {
-      return t.callExpression(methodNode, [targetNode, ...sourceNodes]);
+    ([targetNode, ...sourceNodes]: Array<BabelNodeExpression>) => {
+      return t.callExpression(preludeGenerator.memoizeReference("Object.assign"), [targetNode, ...sourceNodes]);
     },
     { skipInvariant: true }
```

Comment:
Why doesn't this one get `, mutatesOnly: [newState], temporalType: "OBJECT_ASSIGN"`?